packages/forge/src/governance/Governor.sol

```diff
@@ -226,51 +214,87 @@ abstract contract Governor is Context, ERC165, EIP712, IGovernor {
      *
      * Note: Support is generic and can represent various things depending on the voting system used.
      */
-    function _countVote(
-        uint256 proposalId,
-        address account,
-        uint8 support,
-        uint256 numVotes,
-        uint256 totalVotes
-    ) internal virtual;
+    function _countVote(uint256 proposalId, address account, uint8 support, uint256 numVotes, uint256 totalVotes)
+        internal
+        virtual;
+
+    /**
+     * @dev See {IGovernor-verifyProposer}.
+     */
+    function verifyProposer(address account, bytes32[] calldata proof) public override returns (bool verified) {
+        if (!addressSubmitterVerified[account]) {
+            checkProof(account, AMOUNT_FOR_SUMBITTER_PROOF, proof, false); // will revert with NotInMerkle if not valid
+            addressSubmitterVerified[account] = true;
+        }
+        return true;
+    }
+
+    /**
+     * @dev See {IGovernor-validateProposalData}.
+     */
+    function validateProposalData(ProposalCore memory proposal) public virtual override returns (bool dataValidated) {
+        require(proposal.author == msg.sender, "Governor: the proposal author must be msg.sender");
+        for (uint256 index = 0; index < METADATAS_COUNT; index++) {
```

Comment:
Oo this is such helpful gas optimization, thank you! Readability and simplicity has taken the fore so far because our entire thesis is that rollups are the future and so gas optimization is not priority, but these are helpful to consider!

Reply:
You're welcome! The first two don't even alter readability, though I must agree that the third one is a bit ugly. I'm using it as I'm kinda an optimization freak, but I'll leave it up to you ofc

Reply:
Oh no I really appreciate it!! Just wanted to provide context for why we aren't as up to speed/optimized and why these comments are really helpful because it's just not something that I think about as much, so really appreciate it!